sdk/loadtestservice/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncClient.java

```diff
@@ -38,314 +43,322 @@ public final class LoadTestAdministrationAsyncClient {
+                break;
+            case "VALIDATION_NOT_REQUIRED":
+                lroStatus = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                validationStatusEnum = ValidationStatus.VALIDATION_NOT_REQUIRED;
+                break;
+            default:
+                lroStatus = LongRunningOperationStatus.NOT_STARTED;
+                validationStatusEnum = ValidationStatus.NOT_VALIDATED;
+                break;
+        }
+        return Mono.just(new PollResponse<>(lroStatus, validationStatusEnum));
     }
 
     /**
-     * Delete server metrics configuration by its name.
+     * Uploads file and polls the validation status of the uploaded file.
      *
-     * @param name Unique name for server metrics, must be a valid URL character ^[a-z0-9_-]*$.
-     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
-     * @throws HttpResponseException thrown if the request is rejected by server.
-     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
-     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
-     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
-     * @return the {@link Response} on successful completion of {@link Mono}.
+     * @param testId Unique name for load test, must be a valid URL character ^[a-z0-9_-]*$.
+     * @param fileName Unique name for test file with file extension like : App.jmx.
+     * @param body The file content as application/octet-stream.
+     * @param refreshTime The time in seconds to refresh the polling operation.
+     * @param fileUploadRequestOptions The options to configure the file upload HTTP request before HTTP client sends
+     *     it.
+     * @throws ResourceNotFoundException when a test with {@code testId} doesn't exist.
+     * @return A {@link PollerFlux} to poll on and retrieve the validation
+     *     status(NOT_VALIDATED/VALIDATION_SUCCESS/VALIDATION_FAILURE/VALIDATION_INITIATED/VALIDATION_NOT_REQUIRED).
      */
-    @Generated
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Void>> deleteServerMetricsWithResponse(String name, RequestOptions requestOptions) {
-        return this.serviceClient.deleteServerMetricsWithResponseAsync(name, requestOptions);
+    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
+    public PollerFlux<ValidationStatus, BinaryData> beginUploadAndValidate(
+            String testId, String fileName, BinaryData body, int refreshTime, RequestOptions fileUploadRequestOptions) {
+        return new PollerFlux<>(
+                Duration.ofSeconds(refreshTime),
+                (context) -> {
+                    Mono<BinaryData> fileMono =
+                            uploadTestFileWithResponse(testId, fileName, body, fileUploadRequestOptions)
```

Comment:
You might want to make a new `RequestOptions`, set its `Context` to `fileUploadRequestOptions.getContext`, and use it for all following "WithResponse" calls in this LRO.

Reason: You want to pass the "Context", but not other query/header parameters within.

Reply:
Does this context contain the `filetype` query param info as well? Also is passing context yield the same result as passing query param in requestOptions?

Reply:
`RequestOptions` could contains query/header parameter and maybe more.

`Context` only contains a tree of key/value https://azure.github.io/azure-sdk/java_introduction.html#java-service-client-context, which mostly used for telemetry e.g. OpenTelemetry

Reply:
Done